runtime/bundles/org.eclipse.core.expressions/src/org/eclipse/core/expressions/CountExpression.java

```diff
@@ -62,68 +63,86 @@ public CountExpression(String size) {
 
 	private void initializeSize(String size) {
 		if (size == null)
-			size= "*"; //$NON-NLS-1$
+			size = "*"; //$NON-NLS-1$
 		if ("*".equals(size)) //$NON-NLS-1$
-			fMode= ANY_NUMBER;
+			fMode = ANY_NUMBER;
 		else if ("?".equals(size)) //$NON-NLS-1$
-			fMode= NONE_OR_ONE;
+			fMode = NONE_OR_ONE;
 		else if ("!".equals(size)) //$NON-NLS-1$
-			fMode= NONE;
+			fMode = NONE;
 		else if ("+".equals(size)) //$NON-NLS-1$
-			fMode= ONE_OR_MORE;
+			fMode = ONE_OR_MORE;
 		else if (size.charAt(0) == '-' && size.charAt(size.length() - 1) == ')') {
 			try {
 				fMode = LESS_THAN;
 				fSize = Integer.parseInt(size.substring(1, size.length() - 1));
 			} catch (NumberFormatException e) {
-				fMode= UNKNOWN;
+				fMode = UNKNOWN;
+			}
+		} else if (size.charAt(0) == '-' && size.charAt(size.length() - 1) == ']') {
+			try {
+				fMode = LESS_THAN_OR_EQUAL;
+				fSize = Integer.parseInt(size.substring(1, size.length() - 1));
+			} catch (NumberFormatException e) {
+				fMode = UNKNOWN;
 			}
 		} else if (size.charAt(0) == '(' && size.charAt(size.length() - 1) == '-') {
 			try {
 				fMode = GREATER_THAN;
 				fSize = Integer.parseInt(size.substring(1, size.length() - 1));
 			} catch (NumberFormatException e) {
-				fMode= UNKNOWN;
+				fMode = UNKNOWN;
+			}
+		} else if (size.charAt(0) == '[' && size.charAt(size.length() - 1) == '-') {
+			try {
+				fMode = GREATER_THAN_OR_EQUAL;
+				fSize = Integer.parseInt(size.substring(1, size.length() - 1));
+			} catch (NumberFormatException e) {
+				fMode = UNKNOWN;
 			}
 		} else {
 			try {
-				fSize= Integer.parseInt(size);
-				fMode= EXACT;
+				fSize = Integer.parseInt(size);
+				fMode = EXACT;
 			} catch (NumberFormatException e) {
-				fMode= UNKNOWN;
+				fMode = UNKNOWN;
 			}
 		}
 	}
 
 	@Override
 	public EvaluationResult evaluate(IEvaluationContext context) throws CoreException {
-		Object var= context.getDefaultVariable();
+		Object var = context.getDefaultVariable();
 		int size;
 		if (var instanceof Collection) {
-			size= ((Collection<?>)var).size();
+			size = ((Collection<?>) var).size();
```

Comment:
Please replace the cast with an assignment in the condition, i.e.:
```java
if (var instanceof Collection collection) {
	size = collection.size();
```

Reply:
Thank you for for your reply!

I added the code refactoring in the commit. The tests will run again now and if there is no failing test, the PR can be merged.